status-im/status-desktop
# feat: use 1 content-topic for all chats within a community (not to be merged)

### What does the PR do
This PR is mainly to dogfood underlying changes made in status-go.

Reference PR https://github.com/status-im/status-go/pull/5864 which has details. 

This PR uses single content-topic for all community chats to send messages. It still receives messages on all existing content-topics used by previous versions in order to maintain compatibility. 

Details of the discussion are available https://forum.vac.dev/t/status-communities-review-and-proposed-usage-of-waku-content-topics/335/29

### Affected areas

communities

### Impact on end user

What is the impact of these changes on the end user (before/after behaviour)

### How to test

- This PR would require complete testing of communities in relay and light modes.
- It would also require to be tested for backward compatibility with older version of the code.
- Best to check all admin functions as well using this new code
- We also need to validate if community admin is using older code and a member is using new code there are no incompatibilites.
- All chats and community flows such as add member, remove member, add channel, delete channel, change permissions to channels etc

### Risk 

No risk is seen because this change will not break compatibility.

Tick **one**:
